extensions/replication/ReplicationQueuePopulator.js

```diff
@@ -78,6 +78,31 @@ class ReplicationQueuePopulator extends QueuePopulatorExtension {
                      `${queueEntry.getBucket()}/${queueEntry.getObjectKey()}`,
                      JSON.stringify(publishedEntry));
     }
+
+    /**
+     * Filter if the entry is considered a valid master key entry.
+     * There is a case where a single null entry looks like a master key and
+     * will not have a duplicate versioned key. They are created when you have a
+     * non-versioned bucket with objects, and then convert bucket to versioned.
+     * If no new versioned objects are added for given object(s), they look like
+     * standalone master keys. The `isNull` case is undefined for these entries.
+     * Null versions which are objects created after suspending versioning are allowed,
+     * these only have a master object that has an internal versionId and a 'isNull' flag.
+     * @param {ObjectQueueEntry} entry - raw queue entry
+     * @return {Boolean} true if we should filter entry
+     */
+    _entryCanBeReplicated(entry) {
+        const isMaster = isMasterKey(entry.getObjectVersionedKey());
+        // single null entries will have a version id as undefined or null.
+        // do not filter single null entries
+        const isNonVersionedMaster = entry.getVersionId() === undefined;
+        const isNullVersionedMaster = entry.getIsNull();
+        if (isMaster && !isNonVersionedMaster && !isNullVersionedMaster) {
+            this.log.trace('skipping master key entry');
+            return false;
+        }
+        return true;
```

Comment:
Minor readability suggestion, I tend to think that reversing the logic is easier to reason about, because it transforms into a "either of" rather than a combination of three conditions that all have to be true at once (but up to you to change or not):

```suggestion
        if (!isMaster || isNonVersionedMaster || isNullVersionedMaster) {
            return true;
        }
        this.log.trace('skipping master key entry');
        return false;
```